src/Dialogs/FileManager.cpp

```diff
@@ -100,6 +100,23 @@ CanDownload(const FileRepository &repository, const TCHAR *name)
   return FindRemoteFile(repository, name) != nullptr;
 }
 
+static bool
+UpdateAvailable(const FileRepository &repository, const TCHAR *name)
+{
+  const AvailableFile *remote_file = FindRemoteFile(repository, name);
```

Comment:
This whole function is rather expensive for the draw callback - it will be invoked for **every** file **every** time something on the screen gets drawn!
Don't do stuff in the draw callback!

Reply:
You're right that's prob not a good idea. I'll see if I can compute it elsewhere.